lib/container-mapping.js

```diff
@@ -85,63 +86,83 @@ class ContainerMapping {
+                throw new Error("Unable to get docker images: " + error);
+            });
+            core.debug("Finished data collection, starting API calls.");
+            let bearerToken = yield core.getIDToken()
+                .then((token) => { return token; })
+                .catch((error) => {
+                throw new Error("Unable to get token: " + error);
+            });
+            if (!bearerToken) {
+                throw new Error("Empty OIDC token received");
+            }
+            var reportSent = yield this.sendReport(JSON.stringify(reportData), bearerToken, sendReportRetryCount);
+            if (!reportSent) {
+                throw new Error("Unable to send report to backend service");
+            }
+            ;
+        });
+    }
+    execCommand(command, listener) {
+        return __awaiter(this, void 0, void 0, function* () {
+            return exec.exec(command, null, {
                 listeners: {
                     stdout: (data) => {
-                        reportData.dockerImages = reportData.dockerImages.concat(data.toString());
+                        var lines = data.toString().trim().split(os.EOL);
+                        lines.forEach(element => {
+                            listener.push(element);
+                        });
                     }
                 }
+            })
+                .then((exitCode) => {
+                if (exitCode != 0) {
+                    Promise.reject(`Command failed with exit code ${exitCode}`);
+                }
             });
-            core.debug("Finished data collection, starting API calls.");
-            yield this.sendReport(reportData, sendReportRetryCount);
         });
     }
-    sendReport(data, retryCount = 0) {
+    sendReport(data, bearerToken, retryCount = 0) {
         return __awaiter(this, void 0, void 0, function* () {
-            return new Promise((resolve, reject) => __awaiter(this, void 0, void 0, function* () {
-                do {
-                    try {
-                        yield this._sendReport(data);
-                        resolve();
-                        break;
-                    }
-                    catch (error) {
-                        if (retryCount == 0) {
-                            reject('Failed to send report: ' + error);
-                        }
-                        else {
-                            retryCount--;
-                            core.debug(`Retrying API call. Retry count: ${retryCount}`);
-                        }
-                    }
-                } while (retryCount >= 0);
+            core.debug(`attempting to send report: ${data}`);
+            return yield this._sendReport(data, bearerToken)
+                .then(() => {
+                return true;
+            })
+                .catch((error) => __awaiter(this, void 0, void 0, function* () {
+                if (retryCount == 0) {
+                    return false;
```

Comment:
Do you no longer need to log `Failed to send report`? (I know it is logged until 1)